eq-author-api/schema/resolvers/base.js

```diff
@@ -701,6 +719,141 @@ const Resolvers = {
+      publishCommentUpdates(componentId);
+
+      return newComment;
+    },
+    deleteComment: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+
+      const componentComments = questionnaireComments.comments[componentId];
+      if (componentComments) {
+        remove(componentComments, ({ id }) => id === input.commentId);
+        await saveComments(questionnaireComments);
+      }
+      publishCommentUpdates(componentId);
+      return componentComments;
+    },
+    updateComment: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+      const pageComments = questionnaireComments.comments[componentId];
+
+      if (!pageComments) {
+        throw new Error("No comments found");
+      }
+
+      const commentToEdit = pageComments.find(
+        ({ id }) => id === input.commentId
+      );
+      commentToEdit.commentText = input.commentText;
+      commentToEdit.editedTime = new Date();
+      await saveComments(questionnaireComments);
+
+      publishCommentUpdates(componentId);
+
+      return commentToEdit;
+    },
+    createReply: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+
+      const newReply = {
+        id: uuid.v4(),
+        parentCommentId: input.commentId,
+        commentText: input.commentText,
+        userId: ctx.user.id,
+        createdTime: new Date(),
+      };
+      let parentComment = questionnaireComments.comments[componentId].find(
+        ({ id }) => id === input.commentId
+      );
+      if (parentComment) {
+        parentComment.replies.push(newReply);
+      } else {
+        parentComment = [newReply];
+      }
+
+      await saveComments(questionnaireComments);
+
+      publishCommentUpdates(componentId);
+
+      return newReply;
+    },
+    updateReply: async (_, { input }, ctx) => {
+      const { componentId } = input;
+      const questionnaire = ctx.questionnaire;
+      const questionnaireComments = await getCommentsForQuestionnaire(
+        questionnaire.id
+      );
+      const replies = questionnaireComments.comments[componentId].find(
+        ({ id }) => id === input.commentId
+      ).replies;
+
+      if (!replies) {
+        throw new Error("No replies found!");
+      }
+
+      const replyToEdit = replies.find(({ id }) => id === input.replyId);
+      replyToEdit.commentText = input.commentText;
+      replyToEdit.editedTime = new Date();
+      await saveComments(questionnaireComments);
+
+      publishCommentUpdates(componentId);
+      return replyToEdit;
```

Comment:
Same as above reg `input.____`

Reply:
Thanks matt - much cleaner and I've repeated in the comments resolver too!